packages/bff/src/services/authorizationService.ts

```diff
@@ -96,6 +107,168 @@ export function authorizationServiceBuilder(
     },
   });
 
+  const validateSignature = (saml: SAMLResponse): void => {
+    const response = saml.Response;
+    const reference = response?.Signature?.SignedInfo?.Reference;
+    if (!reference) {
+      throw samlNotValid("Missing Signature Reference");
+    }
+    if (reference.URI !== response?.ID) {
+      throw samlNotValid("Reference URI is not compliant");
+    }
+    const transforms = reference.Transforms?.Transform;
+    if (!transforms) {
+      throw samlNotValid("Missing Transforms");
+    }
+    if (transforms.length > 2) {
+      throw samlNotValid("Transforms are not compliant");
+    }
+    const atLeastOneEnvelopedSignature = !transforms.some(
+      (t) =>
+        t.Algorithm === "http://www.w3.org/2000/09/xmldsig#enveloped-signature"
```

Comment:
I have a doubt 🤔  [here](https://github.com/korteke/java-opensaml/blob/master/opensaml-saml-impl/src/main/java/org/opensaml/saml/security/impl/SAMLSignatureProfileValidator.java#L233) in the java lib compare `Transform.Uri` with algorithm constant  `TRANSFORM_ENVELOPED_SIGNATURE`. 
Meanwhile watching the zod object `SAMLResponse` definition the `Transform` have field `Algorithm` and it's align with example of [XML defined in scala project](https://github.com/pagopa/interop-be-backend-for-frontend/blob/327f063053c77373eb574e511f1daf4f7b64b4b6/src/test/resources/saml2.xml#L14) It's not clear, do you have done some test with this XML parsing?